src/main/kotlin/org/jetbrains/research/testspark/display/TestCasePanel.kt

```diff
@@ -344,6 +328,18 @@ class TestCasePanelBuilder(
         })
     }
 
+    private fun updateErrorRelatedUI() {
+        updateBorder(error)
+        if (error.isNullOrBlank()) {
+            errorLabel.isVisible = false
+        } else {
+            errorLabel.toolTipText = error
+            errorLabel.isVisible = true
+        }
+
+        runTestButton.isEnabled = previousError(testCase.testCode) == null
+    }
```

Comment:
The function name `previousError` is not very good. `getPreviousError`/extractPreviousError` is better.